File: c/exemples/exemple_graphe_matrice_adjacence.c
#include "graphe_matrice_adjacence.h"

#include <stdio.h>

int main(void) {
    GrapheMatriceAdjacence graphe = creer_graphe_matrice_adjacence(4, 0);

    ajouter_arete_graphe_matrice_adjacence(&graphe, 0, 1);
    ajouter_arete_graphe_matrice_adjacence(&graphe, 0, 2);
    ajouter_arete_graphe_matrice_adjacence(&graphe, 2, 3);

    printf("Matrice d'adjacence initiale :\n");
    afficher_graphe_matrice_adjacence(&graphe);

    retirer_arete_graphe_matrice_adjacence(&graphe, 0, 2);
    printf("Apres suppression de l'arete entre 0 et 2 :\n");
    afficher_graphe_matrice_adjacence(&graphe);

    detruire_graphe_matrice_adjacence(&graphe);
    return 0;
}

File: ocaml/exemples/exemple_graphe_matrice_adjacence.ml
let chaine_de_liste_entiere valeurs =
  "[" ^ String.concat " ; " (List.map string_of_int valeurs) ^ "]"

let () =
  let graphe = Graphe_matrice_adjacence.creer_graphe_matrice_adjacence 4 false in
  Graphe_matrice_adjacence.ajouter_arete_graphe_matrice_adjacence graphe 0 1;
  Graphe_matrice_adjacence.ajouter_arete_graphe_matrice_adjacence graphe 0 2;
  Graphe_matrice_adjacence.ajouter_arete_graphe_matrice_adjacence graphe 2 3;
  Printf.printf "Voisins de 0 : %s\n"
    (chaine_de_liste_entiere
       (Graphe_matrice_adjacence.voisins_graphe_matrice_adjacence graphe 0));
  Graphe_matrice_adjacence.retirer_arete_graphe_matrice_adjacence graphe 0 2;
  Printf.printf "Voisins de 0 apres suppression : %s\n"
    (chaine_de_liste_entiere
       (Graphe_matrice_adjacence.voisins_graphe_matrice_adjacence graphe 0))
